src/order-book/api.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  Address,
+  BaseHttpRequest,
+  CancelablePromise,
+  DefaultService,
+  NativePriceResponse,
+  OpenAPIConfig,
+  OrderBookClient,
+  OrderCancellation,
+  OrderCreation,
+  OrderPostError,
+  OrderQuoteRequest,
+  OrderQuoteResponse,
+  Trade,
+  TransactionHash,
+  UID,
+} from './generated'
+import { CowError } from '../common/cow-error'
+import { SupportedChainId } from '../common/chains'
+import { EnvConfigs, PROD_CONFIG, STAGING_CONFIG } from '../common/configs'
+import { transformOrder } from './transformOrder'
+import { EnrichedOrder } from './types'
+import { ApiRequestOptions } from './generated/core/ApiRequestOptions'
+import { request as __request } from './generated/core/request'
+
+class FetchHttpRequest extends BaseHttpRequest {
+  constructor(config: OpenAPIConfig) {
+    super(config)
+  }
+
+  /**
+   * Request method
+   * @param options The request options from the service
+   * @returns CancelablePromise<T>
+   * @throws ApiError
+   */
+  public override request<T>(options: ApiRequestOptions): CancelablePromise<T> {
+    return __request(this.config, {
+      ...options,
+      headers: {
+        ...options.headers,
+        'Content-Type': 'application/json',
+      },
+    })
+  }
+}
+
+export class OrderBookApi {
+  private envConfig: EnvConfigs
+  private servicePerNetwork: Record<SupportedChainId, OrderBookClient | null> = {
+    [SupportedChainId.MAINNET]: null,
+    [SupportedChainId.GOERLI]: null,
+    [SupportedChainId.GNOSIS_CHAIN]: null,
+  }
+
+  constructor(env: 'prod' | 'staging' = 'prod') {
+    this.envConfig = env === 'prod' ? PROD_CONFIG : STAGING_CONFIG
+  }
+
+  getTrades(
+    chainId: SupportedChainId,
```

Comment:
Sasha, one thing that I'm not very found of in the new api is that the chainId is the first parameter on all calls and it's required.

We envisioned it to default to whatever the SDK was instantiated with and optionally could be overridden on a call by call basis

Similarly, the env is on the constructor and I don't see an option to change it on individual calls
Nor to say "query both", in the example of getting a single order

Reply:
Discussed, moved back the previous approach (with context)